src/Analyzer/QueryTreeBuilder.cpp

```diff
@@ -1071,4 +1087,42 @@ QueryTreeNodePtr buildQueryTree(ASTPtr query, ContextPtr context)
     return builder.getQueryTreeNode();
 }
 
+QueryTreeNodePtr QueryTreeBuilder::setFirstArgumentAsParameter(const ASTFunction * function, const ContextPtr & context) const
+{
+    const auto * first_arg_ast = function->arguments->children[0].get();
+    const auto * first_arg_literal = first_arg_ast->as<ASTLiteral>();
+
+    if (!first_arg_literal)
+    {
+        throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+            "If groupConcat is used with two arguments, the first argument must be a constant String");
+    }
+
+    if (first_arg_literal->value.getType() != Field::Types::String)
+    {
+        throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+            "If groupConcat is used with two arguments, the first argument must be a constant String");
+    }
+
+    std::string separator = first_arg_literal->value.safeGet<String>();
+
+    ASTPtr second_arg = function->arguments->children[1]->clone();
+
+    auto function_node = std::make_shared<FunctionNode>(function->name);
+    function_node->setNullsAction(function->nulls_action);
+
+    function_node->getParameters().getNodes().push_back(buildExpression(function->arguments->children[0], context)); // Separator
+    function_node->getArguments().getNodes().push_back(buildExpression(second_arg, context)); // Column to concatenate
+
+    if (function->is_window_function)
```

Comment:
I just rewrote it from the original code, where functions were handled. I didn't delete the window functions handling because aggregate functions can be window functions